packages/webdriverio/src/utils.ts

```diff
@@ -98,6 +101,33 @@ export const axeSourceInject = async (
   );
 };
 
+async function assertFrameReady(client: Browser): Promise<void> {
+  // Wait so that we know there is an execution context.
+  // Assume that if we have an html node we have an execution context.
+  try {
+    /*
+      When using the devtools protocol trying to call
+      client.execute() on an unloaded iframe would cause
+      the code to hang indefinitely since it is using
+      Puppeteer which freezes on unloaded iframes. Set a
+      race timeout in order to handle that. Code taken
+      from our @axe-core/puppeteer utils function.
+      @see https://github.com/dequelabs/axe-core-npm/issues/727
+    */
+    const timeoutPromise = new Promise((resolve, reject) => {
+      setTimeout(() => {
+        reject();
+      }, FRAME_LOAD_TIMEOUT)
+    });
+    const executePromise = client.execute(() => {
+      return document.readyState === 'complete'
+    });
+    await Promise.race([timeoutPromise, executePromise]);
+  } catch {
+    throw new Error('Page/Frame is not ready');
```

Comment:
Isn't part of the point of making these changes that we do not throw an exception?

Reply:
We decided that the quickest way to resolve the bug was to return these iframes as `frame-tested` issues. How we want to handle unloaded / lazy loaded iframes is still a topic for discussion and being decided, but that shouldn't slow us down getting a fix out to the client.